src/main/java/org/apache/commons/compress/compressors/CompressorStreamFactory.java

```diff
@@ -522,6 +547,27 @@ public CompressorInputStream createCompressorInputStream(final InputStream in) t
         return createCompressorInputStream(detect(in), in);
     }
 
+    /**
+     * Create a compressor input stream from an input stream, auto-detecting the
+     * compressor type from the first few bytes of the stream while limiting the detected type
+     * to the provided set of compressor names. The InputStream must support marks, like BufferedInputStream.
+     *
+     * @param in
+     *            the input stream
+     * @param compressorNames
+     *            compressor names to limit autodetection
+     * @return the compressor input stream
+     * @throws CompressorException
+     *             if the autodetected compressor is not in the provided set of compressor names
+     * @throws IllegalArgumentException
+     *             if the stream is null or does not support mark
+     * @since 1.25
```

Comment:
The next version will be 1.25.1 but this will be for 1.26.0 since this PR adds a new public API.